src/test/java/ai/nightfall/scan/NightfallClientTest.java

```diff
@@ -0,0 +1,310 @@
+package ai.nightfall.scan;
+
+import ai.nightfall.scan.model.Finding;
+import ai.nightfall.scan.model.NightfallAPIException;
+import ai.nightfall.scan.model.NightfallErrorResponse;
+import ai.nightfall.scan.model.NightfallRequestTimeoutException;
+import ai.nightfall.scan.model.ScanFileRequest;
+import ai.nightfall.scan.model.ScanFileResponse;
+import ai.nightfall.scan.model.ScanPolicy;
+import ai.nightfall.scan.model.ScanTextRequest;
+import ai.nightfall.scan.model.ScanTextResponse;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.mockwebserver.Dispatcher;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import okhttp3.mockwebserver.SocketPolicy;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.migrationsupport.rules.EnableRuleMigrationSupport;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Unit tests for the Nightfall Client.
+ */
+@EnableRuleMigrationSupport
+public class NightfallClientTest {
+
+    @Test
+    public void testScanText_HappyPath() {
+        try (MockWebServer server = new MockWebServer()) {
+            server.enqueue(new MockResponse().setBody("{\"findings\": [[]]}"));
+
+            List<List<Finding>> expectedFindings = Arrays.asList(Arrays.asList());
+            NightfallClient c = new NightfallClient(getRequestURL(server), "key", 1, getHttpClient());
+            ScanTextRequest req = new ScanTextRequest(null, null);
+            ScanTextResponse resp = c.scanText(req);
+            assertEquals(expectedFindings, resp.getFindings());
+        } catch (IOException e) {
+            fail("IOException during test: " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void testScanText_InvalidArgs() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            NightfallClient client = new NightfallClient("host", "key", 1, new OkHttpClient());
```

Comment:
nit: What's the invalid arg in this test? the `"host"`? If so, can you rename it to `"invalid host"` just so it's more obvious.

Reply:
nah it's the `null` passed to `scanText()`, will do